libsolidity/experimental/codegen/IRGeneratorForStatements.cpp

```diff
@@ -205,9 +205,29 @@ void IRGeneratorForStatements::endVisit(BinaryOperation const& _binaryOperation)
 	// TODO: get around resolveRecursive by passing the environment further down?
 	functionType = m_context.env->resolveRecursive(functionType);
 	m_context.enqueueFunctionDefinition(&functionDefinition, functionType);
-	// TODO: account for return stack size
-	m_code << "let " << IRNames::localVariable(_binaryOperation) << " := " << IRNames::function(*m_context.env, functionDefinition, functionType) << "("
-		<< IRNames::localVariable(_binaryOperation.leftExpression()) << ", " << IRNames::localVariable(_binaryOperation.rightExpression()) << ")\n";
+	m_code << "let " << var(_binaryOperation).commaSeparatedList() <<
```

Comment:
Yes, I attempted to do it using fold expressions with variadic templates to have a more generic function for building function calls, however, as `_functionCall.arguments()` is not a constant expression this would just not work, instead, I'm passing all as a vector now. It is not very nice for the `_binaryOperation` case, but it at least removes some of the duplicated code.

Reply:
Solved it passing a vector of arguments, see: https://github.com/ethereum/solidity/pull/14620/commits/5a68eaf54b955ad3f7bb751594262fced0c21171